reporting.md

```diff
@@ -0,0 +1,462 @@
+ 	 * In the case of `mailto`, reports should be submitted to the specified
+     email address. When sending failure reports via SMTP, sending MTAs MUST
+     NOT honor SMTP STS or DANE TLSA failures.
+   * `detailed-report-uri`: Future use. (There may also be a need for enabling
+     more detailed "forensic" reporting during initial stages of a deployment.
+     To address this, the authors consider the possibility of an optional
+     additional "forensic reporting mode" in which more details--such as
+     certificate chains and MTA banners--may be reported. See the section
+     _Future_ _Work_ for more details.)
+
+
+## Example Reporting Policy
+
+### Report using MAILTO:
+
+```_smtp_tlsrpt.mail.example.com. IN TXT \
+		"version:TLSRPT1;aggregate-report-uri:mailto:reports@example.com"
+```
+
+### Report using HTTPS:
+
+```_smtp_tlsrpt.mail.example.com. IN TXT \
+		"version:TLSRPT1; \
+		aggregate-report-uri:https://reporting.example.com/v1/tlsrpt"
+```
+
+# Reporting Schema
+
+Aggregate reports contain the following fields:
+
+* _Report metadata_: 
+	* The organization responsible for the report
+	* Contact information for one or more responsible parties for the contents 
+	of the report
+	* A unique identifier for the report
+	* The reporting date range for the report
+* _Applied policy_, consisting of: 
+	* One of the following policy types:
+		* The SMTP MTA STS policy applied (as a string)
+		* The DANE TLSA record applied (as a string)
+	* The domain for which the policy is applied
+	* The MX host
+	* An identifier for the policy (where applicable)
+* _Aggregate counts_, comprising _result type_, _sending MTA IP_, _receiving
+  MTA hostname_, _message count_, and an optional _additional information_
+  field containing a URI for recipients to review further information on a
+  failure type.
+
+
+
+Note that the failure types are non-exclusive; an aggregate report MAY contain
+overlapping `counts` of failure types where a single send attempt encountered
+multiple errors.
+
+
+## Result Types
+
+The list of result types will start with the minimal set below, and is expected
+  to grow over time based on real-world experience. The initial set is:
+
+### Success Type
+  * `success`: This indicates that the sending MTA was able to successfully
+    negotiate a policy-compliant TLS connection, and serves to provide a
+    "heartbeat" to receiving domains that reporting is functional and
+    tabulating correctly.
+
+  
+### Routing Failures
+  * `mx-mismatch`: This indicates that the MX resolved for the recipient domain
+    did not match the MX constraint specified in the policy.
+  * `certificate-mismatch`: This indicates that the certificate presented by the
+    receiving MX did not match the MX hostname
+
+### Negotiation Failures
+
+  * `starttls-not-supported`: This indicates that the recipient MX did not
+    support STARTTLS.
+  * `invalid-certificate`: This indicates that the certificate presented by the
```

Comment:
there are 2 categories with invalid-certificate:
1. Invalid certs (in other words, the cert is not trusted)
2. Cert is trusted but with a name mismatch.